# HG changeset patch
# User Fabian Groffen <grobian@gentoo.org>
# Date 1464576157 -3600
#      Mon May 30 03:42:37 2016 +0100
# Node ID 83425b86fe900b4df0c21408413462c65ded6f32
# Parent  684370c57b850b13c7d6c5da32f8d37f6f40886c
[PATCH] dont-reveal-bbc
http://bugs.debian.org/cgi-bin/bugreport.cgi?msg=28;filename=write_bcc.patch.1.
5.20-2;att=1;bug=467432

originally adapted for Gentoo patchsets, this should be identical to
debian/0011-debian-specific-467432-write_bcc.patch however, it is NOT.
There is a slight comment difference, and a write header call
difference.

additional information:
http://dev.mutt.org/trac/ticket/3337
http://bugs.debian.org/467432
---
 headers.c | 2 +-
 init.h    | 8 +++-----
 main.c    | 2 +-
 protos.h  | 2 +-
 send.c    | 4 ++--
 sendlib.c | 6 +++---
 6 files changed, 11 insertions(+), 13 deletions(-)

diff -r 684370c57b85 -r 83425b86fe90 PATCHES
--- a/PATCHES	Mon May 30 03:42:21 2016 +0100
+++ b/PATCHES	Mon May 30 03:42:37 2016 +0100
@@ -1,3 +1,4 @@
+patch-dont-reveal-bcc-gentoo (modified version from Debian)
 patch-multiple-fcc-neomutt
 patch-timeout-neomutt
 patch-new-mail-neomutt
diff -r 684370c57b85 -r 83425b86fe90 headers.c
--- a/headers.c	Mon May 30 03:42:21 2016 +0100
+++ b/headers.c	Mon May 30 03:42:37 2016 +0100
@@ -53,7 +53,7 @@ void mutt_edit_headers (const char *edit
   }
   
   mutt_env_to_local (msg->env);
-  mutt_write_rfc822_header (ofp, msg->env, NULL, 1, 0);
+  mutt_write_rfc822_header (ofp, msg->env, NULL, 1, 0, 1);
   fputc ('\n', ofp);	/* tie off the header. */
 
   /* now copy the body of the message. */
diff -r 684370c57b85 -r 83425b86fe90 init.h
--- a/init.h	Mon May 30 03:42:21 2016 +0100
+++ b/init.h	Mon May 30 03:42:37 2016 +0100
@@ -4216,11 +4216,9 @@ struct option_t MuttVars[] = {
   { "write_bcc",	DT_BOOL, R_NONE, OPTWRITEBCC, 1},
   /*
   ** .pp
-  ** Controls whether mutt writes out the ``Bcc:'' header when preparing
-  ** messages to be sent.  Exim users may wish to unset this. If mutt
-  ** is set to deliver directly via SMTP (see $$smtp_url), this
-  ** option does nothing: mutt will never write out the ``Bcc:'' header
-  ** in this case.
+  ** Controls whether mutt writes out the Bcc header when saving
+  ** messages to FCC. Bcc headers will never be written to a message
+  ** when sending it.
   */
   { "write_inc",	DT_NUM,	 R_NONE, UL &WriteInc, 10 },
   /*
diff -r 684370c57b85 -r 83425b86fe90 main.c
--- a/main.c	Mon May 30 03:42:21 2016 +0100
+++ b/main.c	Mon May 30 03:42:37 2016 +0100
@@ -930,7 +930,7 @@ int main (int argc, char **argv, char **
           mutt_env_to_intl (msg->env, NULL, NULL);
         }
 
-        mutt_write_rfc822_header (fout, msg->env, msg->content, -1, 0);
+        mutt_write_rfc822_header (fout, msg->env, msg->content, -1, 0, 0);
         if (option (OPTRESUMEEDITEDDRAFTFILES))
           fprintf (fout, "X-Mutt-Resume-Draft: 1\n");
         fputc ('\n', fout);
diff -r 684370c57b85 -r 83425b86fe90 protos.h
--- a/protos.h	Mon May 30 03:42:21 2016 +0100
+++ b/protos.h	Mon May 30 03:42:37 2016 +0100
@@ -392,7 +392,7 @@ int mutt_write_multiple_fcc (const char 
 int mutt_write_mime_body (BODY *, FILE *);
 int mutt_write_mime_header (BODY *, FILE *);
 int mutt_write_one_header (FILE *fp, const char *tag, const char *value, const char *pfx, int wraplen, int flags);
-int mutt_write_rfc822_header (FILE *, ENVELOPE *, BODY *, int, int);
+int mutt_write_rfc822_header (FILE *, ENVELOPE *, BODY *, int, int, int);
 void mutt_write_references (LIST *, FILE *, int);
 int mutt_yesorno (const char *, int);
 void mutt_set_header_color(CONTEXT *, HEADER *);
diff -r 684370c57b85 -r 83425b86fe90 send.c
--- a/send.c	Mon May 30 03:42:21 2016 +0100
+++ b/send.c	Mon May 30 03:42:37 2016 +0100
@@ -1083,10 +1083,10 @@ static int send_message (HEADER *msg)
     unset_option (OPTWRITEBCC);
 #endif
 #ifdef MIXMASTER
-  mutt_write_rfc822_header (tempfp, msg->env, msg->content, 0, msg->chain ? 1 : 0);
+  mutt_write_rfc822_header (tempfp, msg->env, msg->content, 0, msg->chain ? 1 : 0, 0);
 #endif
 #ifndef MIXMASTER
-  mutt_write_rfc822_header (tempfp, msg->env, msg->content, 0, 0);
+  mutt_write_rfc822_header (tempfp, msg->env, msg->content, 0, 0, 0);
 #endif
 #ifdef USE_SMTP
   if (old_write_bcc)
diff -r 684370c57b85 -r 83425b86fe90 sendlib.c
--- a/sendlib.c	Mon May 30 03:42:21 2016 +0100
+++ b/sendlib.c	Mon May 30 03:42:37 2016 +0100
@@ -1998,7 +1998,7 @@ out:
 
 
 int mutt_write_rfc822_header (FILE *fp, ENVELOPE *env, BODY *attach,
-			      int mode, int privacy)
+			      int mode, int privacy, int should_write_bcc)
 {
   char buffer[LONG_STRING];
   char *p, *q;
@@ -2047,7 +2047,7 @@ int mutt_write_rfc822_header (FILE *fp, 
 #endif
     fputs ("Cc: \n", fp);
 
-  if (env->bcc)
+  if (env->bcc && should_write_bcc)
   {
     if(mode != 0 || option(OPTWRITEBCC))
     {
@@ -2931,7 +2931,7 @@ int mutt_write_fcc (const char *path, HE
   /* post == 1 => postpone message. Set mode = -1 in mutt_write_rfc822_header()
    * post == 0 => Normal mode. Set mode = 0 in mutt_write_rfc822_header()
    * */
-  mutt_write_rfc822_header (msg->fp, hdr->env, hdr->content, post ? -post : 0, 0);
+  mutt_write_rfc822_header (msg->fp, hdr->env, hdr->content, post ? -post : 0, 0, 1);
 
   /* (postponment) if this was a reply of some sort, <msgid> contians the
    * Message-ID: of message replied to.  Save it using a special X-Mutt-
